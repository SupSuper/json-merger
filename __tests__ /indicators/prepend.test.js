const jsonMerger = require("../../dist");
const {testConfig} = require("../__helpers__");

describe("when merging two arrays and the source item has", function () {

    test("$prepend it should prepend the source array item", function () {

        const object1 = {
            "a": [
                {
                    "aa": "string"
                }
            ]
        };

        const object2 = {
            "a": [
                {
                    "$prepend": true,
                    "bb": "prepend"
                }
            ]
        };

        const result = jsonMerger.mergeObjects([object1, object2], testConfig());

        expect(result).toMatchSnapshot();
    });

    test("$prepend and an other item also has $prepend it should prepend them in the order they are defined", function () {

        const object1 = {
            "a": [
                {
                    "aa": "string"
                }
            ]
        };

        const object2 = {
            "a": [
                {
                    "$prepend": true,
                    "bb": "prepend 1"
                },
                {
                    "$prepend": true,
                    "cc": "prepend 2"
                }
            ]
        };

        const result = jsonMerger.mergeObjects([object1, object2], testConfig());

        expect(result).toMatchSnapshot();
    });

    test("$prepend it should prepend the source array item even if there is an $index indicator with index: 0", function () {

        const object1 = {
            "a": [
                {
                    "aa": "string"
                }
            ]
        };

        const object2 = {
            "a": [
                {
                    "$prepend": true,
                    "bb": "prepend"
                },
                {
                    "$insert": 0,
                    "cc": "insert"
                }
            ]
        };

        const result = jsonMerger.mergeObjects([object1, object2], testConfig());

        expect(result).toMatchSnapshot();
    });
});

test("$prepend on a non array item should do nothing and be stripped", function () {

    const object1 = {
        "1a": {
            "1aa": 1
        }
    };

    const object2 = {
        "2b": {
            "$prepend": true,
            "2bb": 2
        }
    };

    const result = jsonMerger.mergeObjects([object1, object2], testConfig());

    expect(result).toMatchSnapshot();
});
