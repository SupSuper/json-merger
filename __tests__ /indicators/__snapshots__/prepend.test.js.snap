// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`$prepend on a non array item should do nothing and be stripped 1`] = `
"{
	\\"1a\\": {
		\\"1aa\\": 1
	},
	\\"2b\\": {
		\\"2bb\\": 2
	}
}"
`;

exports[`when merging two arrays and the source item has $prepend and an other item also has $prepend it should prepend them in the order they are defined 1`] = `
"{
	\\"a\\": [
		{
			\\"bb\\": \\"prepend 1\\"
		},
		{
			\\"cc\\": \\"prepend 2\\"
		},
		{
			\\"aa\\": \\"string\\"
		}
	]
}"
`;

exports[`when merging two arrays and the source item has $prepend it should prepend the source array item 1`] = `
"{
	\\"a\\": [
		{
			\\"bb\\": \\"prepend\\"
		},
		{
			\\"aa\\": \\"string\\"
		}
	]
}"
`;

exports[`when merging two arrays and the source item has $prepend it should prepend the source array item even if there is an $index indicator with index: 0 1`] = `
"{
	\\"a\\": [
		{
			\\"bb\\": \\"prepend\\"
		},
		{
			\\"cc\\": \\"insert\\"
		},
		{
			\\"aa\\": \\"string\\"
		}
	]
}"
`;
